src/AutoRest.CSharp/Common/Generation/Writers/CodeWriterExtensions.cs

```diff
@@ -257,7 +261,7 @@ public static void WriteVariableAssignmentWithNullCheck(this CodeWriter writer,
         {
             // Temporary check to minimize amount of changes in existing generated code
             var assignToSelf = parameter.Name == variableName;
-            if (parameter.DefaultValue != null && !TypeFactory.CanBeInitializedInline(parameter.Type, parameter.DefaultValue))
+            if (parameter.DefaultValue != null && (!parameter.UseDefaultValueInCtorParam || !TypeFactory.CanBeInitializedInline(parameter.Type, parameter.DefaultValue)))
```

Comment:
Can we change the name to something like "SetDefaultValueInBody"? In that case, property will have meaning in any signature, not just constructor

Reply:
then the property doesn't make sense in the other place its used where we skip setting the actual default in the ctor param.

I can make two properties if you want one with each name and use them in each place, but the usage here should always be the opposite of the usage in the other place

Reply:
Why? It perfectly makes sense to me:
```cs
if (clientParameter.DefaultValue != null && !clientParameter.SetDefaultValueInBody) 
{
```

`WriteParameter` writes parameter **in the signature** of the method. If parameter has default value, but `SetDefaultValueInBody: true`, default value must be set **in the body**. Hence, just `= default` in the signature.